pkg/adapter/metrics.go

```diff
@@ -69,7 +69,19 @@ type (
 		Builder
 
 		// NewMetrics returns a new instance of the Metrics aspect.
-		NewMetrics(env Env, config AspectConfig) (MetricsAspect, error)
+		NewMetrics(env Env, config AspectConfig, metrics []MetricDefinition) (MetricsAspect, error)
+	}
+
+	// MetricDefinition provides the basic description of a metric schema
+	// for which metrics adapters will be sent Values at runtime.
+	MetricDefinition struct {
+		// Name is the canonical name of the metric.
+		Name string
+		// Kind provides type information about the metric.
+		Kind Kind
+		// Labels are the names of keys for dimensional data that will
+		// generated at runtime and passed along with metric values.
+		Labels []string
```

Comment:
Zack, labels are created by the user via descriptors. Individual descriptors specify the set of labels in terms of name/type (so "user_name" or type "string", or "provider_url" of type "Url").

See issue #201 for more context.